### PR TITLE
ticket 8.1

### PR DESCRIPTION
**Pull request by charli353**

### CODE REVIEW

File: src/components/Article.jsx [hunk 1]
```diff
@@ -60,6 +60,11 @@ export default function (props) {
         <button id='commentsbutton' onClick={(event) => {
           setShow(prevShow => !prevShow)
         }}><ButtonSwitch show={showComments}/></button>
+        <LeaveComment show={showComments} id={currentId} onSubmit={(event) => {
+          getComments(currentId).then((response) => {
```
Comment (Beth-McCormack) [inline src/components/Article.jsx]: In this onSubmit, you shouldn't be making an api call, this should be happening in a useEffect. It is more efficiently to only reset the comments in the onSubmit.

**Review by Beth-McCormack — CHANGES_REQUESTED** [review]
Please see comment I've left and resubmit when this is done🙂

File: src/componentfuncs/Comments.jsx [hunk 2]
```diff
@@ -19,4 +23,56 @@ export function RetrieveComments (props) {
...
   
+export function LeaveComment (props) {
+  const [comment, setComment] = useState()
+  const [validity, setValid] = useState(true)
+  const [inputFeedback, setInputFeedback] = useState('success')
+
+
+  const showing = props.show
+  const currentId = props.id
+
```
Comment (sirKevlar) [inline src/componentfuncs/Comments.jsx:34]: This is a strange variable name (commentValidate)... Is this function's only job to validate a comment?

File: src/componentfuncs/Comments.jsx [hunk 3]
```diff
@@ -19,4 +23,56 @@ export function RetrieveComments (props) {
...
+  const commentValidate = (comment) => {
+    if (typeof comment === 'string' && comment.length > 15){
+      postComment(currentId, comment)
```
Comment (sirKevlar) [inline src/componentfuncs/Comments.jsx]: What happens if the post fails? In order to complete this ticket, we need to handle the possibility of error upon posting

File: src/componentfuncs/Comments.jsx [hunk 4]
```diff
@@ -19,4 +23,56 @@ export function RetrieveComments (props) {
...
+  
+  if(showing){
+    return (
+      <div id='commentarea'>
+        <form>
+          <label htmlFor={inputFeedback}></label>
+          <input type="text" id={inputFeedback} value={comment} placeholder='Post a Comment' onChange={(event) => {
+            setComment(event.target.value)
+          }}/>
+          <FailedForm valid={validity}/>
```
Comment (sirKevlar) [inline src/componentfuncs/Comments.jsx]: An onClick here isn't good for accessibility reasons. ATM only mouse users can use your posting function. How could we make this accessible to keyboard users (maybe a listener on the form would help?). We need to change this to complete this ticket please

File: src/componentfuncs/Comments.jsx [hunk 5]
```diff
@@ -19,4 +23,56 @@ export function RetrieveComments (props) {
...
+            commentValidate(comment)
+            setComment('')
+          }}>Submit</button>
+        </form>
+      </div>
+    )
+  }
+}
+
+function FailedForm (props) {
```
Comment (sirKevlar) [inline src/componentfuncs/Comments.jsx]: Failed implies an error. I think this func should be called invalidForm

File: src/App.css [hunk 6]
```diff
@@ -174,7 +203,12 @@
 }
 
 .trendall:focus {
-  border: 2px #F4EEE0 solid;
+  border: 2px #b73840 solid;
+}
+
+
+header {
+  color: #F4EEE0;
```
Comment (sirKevlar) [inline src/App.css:247]: The responsive design on this app is a little strange. In larger views all of the content is squeezed in to the first third of the screen with the other 2 thirds empty. Also the comments section is very easy to break by squeezing screen size. I think you should play with the responsiveness to check all screen sizes work

File: src/componentfuncs/Comments.jsx [hunk 7]
```diff
@@ -19,4 +23,62 @@ export function RetrieveComments (props) {
...
+      setValid(false)
+    }
+  }
+
+
+
+  
+  if(showing){
+    return (
+      <div id='commentarea'>
```
Comment (RichardNeat) [inline src/componentfuncs/Comments.jsx]: Could we use any better html tag here than a div? And anywhere else you may have used a div.

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice work so far on this one. Just a few points to address please:
- I am getting an error when I begin a typing a new comment, could you take a look and correct please. Currently this means I can't see my comment being posted, nor when it refreshes. Make sure this comment renders at the top of the list when submitted.
- Consider what happens on a slow connection to make sure the user cannot click submit multiple times.
- Ensure that the error handling functions as expected.
- Please avoid using divs

Have a look at these and then resubmit when you're ready.

File: src/components/Article.jsx [hunk 8]
```diff
@@ -38,14 +38,14 @@ export default function (props) {
 
 
   return loading ? <p className='loader'>Loading...</p> : (
-    <div id='singleArticle'>
+    <section id='singleArticle'>
       <div id='articletitle'>
       <h2>"{article.title}"</h2>
       </div>
-      <div id='articleimg'>
+      <text id='articleimg'>
```
Comment (mezzDavies) [inline src/components/Article.jsx:45]: I'm a bit confused by this `<text>` tag. It is currently leading to an error - "The tag `<text>` is unrecognized in this browser." Could you just add the id to the `img` tag instead and do without the `<text>` one (is it actually a tag 🤷 😄 )

File: src/componentfuncs/Comments.jsx [hunk 9]
```diff
@@ -32,9 +32,11 @@ export function LeaveComment (props) {
   const showing = props.show
   const currentId = props.id
 
-  const commentValidate = (comment) => {
+  const formValidation = (comment) => {
     if (typeof comment === 'string' && comment.length > 15){
-      postComment(currentId, comment)
+      postComment(currentId, comment).catch((err) => {
+        setValid('Network Error, Try Again')
```
Comment (mezzDavies) [inline src/componentfuncs/Comments.jsx:38]: Although this is working you are changing the data type of validity here. Its initial state is a boolean of true and you set it's state to false elsewhere. Changing it to a string is is not good practice. I would strongly suggest you find another way to get the error message rendered on screen.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
You are so close. It works! But I cannot let it pass as you are using a boolean in a very inadvisable way. Please see my comment below and read the notes [here](https://notes.northcoders.com/courses/js-front-end/error-handling) and I think you'll be able to fix it 👍

**Pull request merged**